gatsby-node.js

```diff
@@ -1,44 +1,56 @@
-exports.createPages = async ({ actions, graphql, reporter }) => {
-  const { createPage, createRedirect } = actions;
-  const emailComponent = require.resolve("./src/components/email/Email.tsx");
+exports.createPages = async ({actions, graphql, reporter}) => {
+  const {createPage, createRedirect} = actions;
 
-  const emailData = await graphql(`
-    {
-      allMarkdownRemark(filter: { frontmatter: { layout: { eq: "email" } } }) {
-        edges {
-          node {
-            frontmatter {
-              permalink
-              redirect_from
+  /** Shared logic between creating email and letter pages */
+  async function createPagesHelper({layout, component}) {
+    const emailData = await graphql(`
```

Comment:
renamed to createEmailOrLetterTemplatePage, renamed emailData to templateData

this file didn't have any types because it was .js, I converted it to .ts wooo